File: src/Model/GeneralStatement/AssignmentStatement.java
package Model.GeneralStatement;

import Model.Expression.IExpression;
import Model.ProgramState;
import Model.Utils.IDictionary;

public class AssignmentStatement implements IStatement {

    private String id;
    private IExpression expression;

    public AssignmentStatement(String id, IExpression expression) {
        this.id = id;
        this.expression = expression;
    }

    @Override
    public ProgramState execute(ProgramState state) {
        IDictionary<String, Integer> symTable = state.getSymTable();
        int value = expression.eval(symTable, state.getHeap());
        symTable.setValue(id, value);
        return null;
    }

    @Override
    public String toString() {
        return id + "=" + expression + ";";
    }
}

File: Toy-Language-C#-Intepreter/Toy-Language-CS-Interpreter/Toy-Language-CS-Interpreter/Model/Statement/GeneralStatement/AssignmentStatement.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Toy_Language_CS_Interpreter.Model.Expression;

namespace Toy_Language_CS_Interpreter.Model.Statement.GeneralStatement
{
    class AssignmentStatement : IStatement
    {
        private string variableName;
        private IExpression expression;

        public AssignmentStatement(string variableName, IExpression expression)
        {
            this.variableName = variableName;
            this.expression = expression;
        }

        public ProgramState Execute(ProgramState state)
        {
            int result = expression.Evaluate(state.SymbolTable);
            state.SymbolTable.Put(variableName, result);
            return state;
        }

        public override string ToString()
        {
            return variableName + "=" + expression + ";";
        }

    }
}
